### ISSUE TITLE
attempting to modify ConfigSync on non-existing device - device not resolving properly

### ISSUE THREAD

**Issue by jgruberf5** [F5Networks/f5-declarative-onboarding#113]
### Environment
 * Declarative Onboarding Version: 1.10.0-2
 * BIG-IP Version: 14.1.2-0.0.37.0

### Summary
Can not provision ConfigSync when device name does not match the hostname and hostname is not explicitly set in the declaration.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
   "schemaVersion" : "1.0.0",
   "async" : true,
   "Common" : {
      "HA-self" : {
         "address" : "1.1.1.108/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "all",
         "vlan" : "HA"
      },
      "poolLicense" : {
         "bigIpUsername" : "admin",
         "reachable" : true,
         "licensePool" : "BIGIPVEREGKEYS",
         "bigIqHost" : "172.13.1.108",
         "bigIqPassword" : "admin",
         "licenseType" : "licensePool",
         "bigIqUsername" : "admin",
         "class" : "License",
         "bigIpPassword" : "f5c0nfig"
      },
      "configsync" : {
         "configsyncIp" : "/Common/HA-self/address",
         "class" : "ConfigSync"
      },
      "HA" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : 1.1
            }
         ],
         "class" : "VLAN",
         "mtu" : 1450
      },
      "failoverAddress" : {
         "address" : "/Common/HA-self/address",
         "class" : "FailoverUnicast"
      },
      "default" : {
         "network" : "default",
         "gw" : "192.168.80.1",
         "class" : "Route",
         "mtu" : 1450
      },
      "external" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : 1.3
            }
         ],
         "class" : "VLAN",
         "mtu" : 1450
      },
      "ntpServers" : {
         "servers" : [
            "0.pool.ntp.org",
            "1.pool.ntp.org",
            "2.pool.ntp.org"
         ],
         "class" : "NTP"
      },
      "dbvars" : {
         "ui.advisory.enabled" : true,
         "ui.advisory.color" : "orange",
         "class" : "DbVariables",
         "ui.advisory.text" : "This device is under centralized management."
      },
      "internal-self" : {
         "address" : "192.168.40.115/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "default",
         "vlan" : "internal"
      },
      "dnsServers" : {
         "search" : [
            "example.openstack.com"
         ],
         "class" : "DNS",
         "nameServers" : [
            "192.168.0.2"
         ]
      },
      "class" : "Tenant",
      "internal" : {
         "interfaces" : [
            {
               "tagged" : false,
               "name" : 1.2
            }
         ],
         "class" : "VLAN",
         "mtu" : 1450
      },
      "external-self" : {
         "address" : "192.168.80.102/24",
         "class" : "SelfIp",
         "trafficGroup" : "traffic-group-local-only",
         "allowService" : "none",
         "vlan" : "external"
      },
      "provisioningLevels" : {
         "ltm" : "nominal",
         "class" : "Provision"
      }
   },
   "class" : "Device",
   "label" : "Cloudinit Onboarding"
}
```

2. Observe the following error response from /var/log/restnoded/restnoded.log
```
Thu, 06 Feb 2020 21:21:00 GMT - finer: [f5-declarative-onboarding: systemHandler.js] tryUntil error: 01020036:3: The requested system device (/Common/adc4nic.openstack.grubernet.org) was not found. tries left: 90
```
Note from restnoded log... repeasted until timeout.

``
Thu, 06 Feb 2020 21:20:57 GMT - fine: [f5-declarative-onboarding: dscHandler.js] Processing DSC declaration.
Thu, 06 Feb 2020 21:20:57 GMT - fine: [f5-declarative-onboarding: dscHandler.js] Checking ConfigSync.
...
**Thu, 06 Feb 2020 21:21:00 GMT - fine: [f5-declarative-onboarding: restWorker.js] modify localhost /tm/cm/device/~Common~adc4nic.openstack.grubernet.org {"configsyncIp":"1.1.1.108"}**
Thu, 06 Feb 2020 21:21:00 GMT - finest: [f5-declarative-onboarding: systemHandler.js] tryUntil: got error {"code":404}
Thu, 06 Feb 2020 21:21:00 GMT - finest: [f5-declarative-onboarding: systemHandler.js] typeof err object
Thu, 06 Feb 2020 21:21:00 GMT - finer: [f5-declarative-onboarding: systemHandler.js] tryUntil error: 01020036:3: The requested system device (/Common/adc4nic.openstack.grubernet.org) was not found. tries left: 90
...
....
Thu, 06 Feb 2020 21:31:23 GMT - finer: [f5-declarative-onboarding: systemHandler.js] tryUntil error: 01020036:3: The requested system device (/Common/adc4nic.openstack.grubernet.org) was not found. tries left: 28
``
From tmsh:

```
[root@adc4nic:Active:Standalone] config # tmsh list cm device
cm device **bigip1** {
    active-modules { "BIG-IP, VE, LAB|ILDHVIR-ADSZOIR|Rate Shaping|External Interface and Network HSM, VE|SSL, Forward Proxy, VE|BIG-IP VE, Multicast Routing|Routing Bundle, VE|ASM, VE|SSL, VE|DNS VE Lab  (10K QPS)|Max Compression, VE|Advanced Protocols, VE|SSL Orchestrator, VE|Advanced Web Application Firewall, VE Lab|AFM, VE (LAB ONLY - NO ROUTING)|DNSSEC|VE, Carrier Grade NAT (AFM ONLY)|PSM, VE|APM, Lab, VE" }
    base-mac fa:16:3e:ce:72:87
    build 0.0.37
    cert dtdi.crt
    chassis-id 17a751b8-f4b1-4942-47df90fbe1d2
    edition Final
    failover-state active
    hostname **adc4nic.openstack.grubernet.org**
    key dtdi.key
    management-ip 192.168.245.105
    marketing-name "BIG-IP Virtual Edition"
    optional-modules { "Anti-Bot Mobile, VE 25 Mbps" "App Mode (TMSH Only, No Root/Bash)" "FIPS 140-2 Level 1, BIG-IP VE-200M" "IP Intelligence, 1Yr, VE-200M/VE-25M" "IP Intelligence, 3Yr, VE-200M/VE-25M" ONAP "Threat Campaigns, 1Yr, VE-200M/VE-25M" "URL Filtering, 1Yr, VE" "URL Filtering, 3Yr, VE" "URL Filtering, VE-25M-1G, 500 Sessions, 1Yr" "URL Filtering, VE-25M-1G, 500 Sessions, 3Yr" }
    platform-id Z100
    product BIG-IP
    self-device true
    time-zone America/Los_Angeles
    version 14.1.2
}
```
You'll never be able to modify a device which doesn't exist.. 

### Expected Behavior
provision ConfigSync to the referenced SelfIp... regardless of the device name.

### Actual Behavior
the request to modify the configsyncip is being placed on a non-existing device because we don't bother to resolve device names properly.

**Comment by jgruberf5** [comment]
Notice in the declaration there was not 'remote' being used, so why wouldn't the device name resolved from /mgmt/shared/identified-devices/config/device-info mcpDeviceName attribute?

BIG FACT OF TMOS CMI: device does not equal hostname ...

**Comment by jgruber** [comment]
I've gotten around this for now by resolving the hostname from the IaaS and then forcing the 'hostname' attribute in the DO declaration. We need to fix device name resolution ASAP.

**Comment by dstokesf5** [comment]
How is this issue different from #104 ?

**Comment by jgruberf5** [comment]
Nothing if you fix it right...

One is an example of setting ConfigSync the other is DeviceGroup. You will need to assure both places in the declaration work with whatever resolution for determining device name (and possibly uniqueness) you choose.

Sent from my T-Mobile 4G LTE Device
Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by dstokesf5** [comment]
Thanks. This should also be covered by AUTOTOOL-1125.

**Comment by dstokesf5** [comment]
This was resolved in the 1.11.0 release.
